src/Context/GtmContext.php

```diff
@@ -51,10 +51,26 @@ public function waitDataLayerSetting($key, $loops = 10) {
    * @Given google tag manager data layer setting :arg1 should be :arg2
    */
   public function dataLayerSettingShouldBe($key, $value) {
-    $property_value = $this->getDataLayerValue($key);
-    if ($value != $property_value) {
-      throw new \Exception($value . ' is not the same as ' . $property_value);
-    }
+      $json_arr = $this->getDataLayerJson();
+      // Loop through the array and return the data layer value
```

Comment:
This code is duplicated, see getDataLayerValue().
There is no need to duplicate the code.